### PR TITLE
Add tests for http client

### LINKED ISSUE #109: MarquezClient has no _now_ms attribute
When running the example usage in the README I get:

AttributeError: 'MarquezClient' object has no attribute '_now_ms'

It's coming from line 353 in client.py. It looks like the client should be using Utils.now_ms() instead of self._now_ms().  I'm happy to make this correction if that's the case.

### PR DESCRIPTION
**Pull request by wslulciuc**
This PR updates the test suite for the HTTP client and fixes numerous bugs. This PR also introduces the use of `DatasetId` and `JobId`.

Fixes #109

### CODE REVIEW

File: marquez_client/client.py [hunk 1]
```diff
@@ -177,30 +178,30 @@ def tag_dataset(self, namespace_name, dataset_name, tag_name):
...
         Utils.check_name_length(job_name, 'job_name')
         Utils.is_instance_of(job_type, JobType)
 
         payload = {
-            'inputs': input_dataset or [],
-            'outputs': output_dataset or [],
-            'type': job_type.name
+            'type': job_type.value,
+            'inputs': [input.__dict__ for input in inputs],
+            'outputs': [output.__dict__ for output in outputs]
```
Comment (julienledem) [inline marquez_client/client.py]: does this handles the None case?

File: tests/test_marquez_client.py [hunk 2]
```diff
@@ -10,483 +10,565 @@
...
+        NAMESPACE_NAME,
+        OWNER_NAME,
+        DESCRIPTION
+    )
+
+    assert namespace['name'] == NAMESPACE_NAME
+    assert namespace['ownerName'] == OWNER_NAME
+    assert namespace['description'] == DESCRIPTION
+
+    mock_put.get.called_once()
```
Comment (julienledem) [inline tests/test_marquez_client.py]: I think you want to use `assert_called_once_with` here.
Something along the lines of:
https://github.com/MarquezProject/marquez-python/blob/3995cce0b71cfa9077a878836ec77e351ced38ee/tests/test_marquez_wo_client.py#L32

Comment (julienledem) [reply tests/test_marquez_client.py]: same comment for other calls bellow

**Review by julienledem — CHANGES_REQUESTED** [review]
This looks great. Code coverage 💯 
To fully verify the behavior you need to call `assert_called_once_with` on the mock insert of just `assert_called_once`.

File: tests/test_marquez_client.py (hunk 2 above)
Comment (wslulciuc) [reply tests/test_marquez_client.py]: Yeah, I'll switch the assert. I was thinking maybe asserting the params was less important in the overall test. But, wouldn't hurt.

File: marquez_client/client.py (hunk 1 above)
Comment (wslulciuc) [reply marquez_client/client.py]: Actually, it doesn't. Nice catch.

**Review by julienledem — APPROVED** [review]
Looking good.
Optionally you can add coverage for the "mark_job_run_as_..." methods

**Comment by wslulciuc** [comment]
> Optionally you can add coverage for the "mark_job_run_as_..." methods

Yep, they've been added 👍

**Pull request merged**